src/java/org/apache/cassandra/cql3/constraints/UnaryFunctionColumnConstraint.java

```diff
@@ -0,0 +1,163 @@
+import org.apache.cassandra.schema.ColumnMetadata;
+import org.apache.cassandra.tcm.serialization.MetadataSerializer;
+import org.apache.cassandra.tcm.serialization.Version;
+import org.apache.cassandra.utils.LocalizeString;
+
+import static org.apache.cassandra.cql3.constraints.ColumnConstraint.ConstraintType.UNARY_FUNCTION;
+
+public class UnaryFunctionColumnConstraint implements ColumnConstraint<UnaryFunctionColumnConstraint>
+{
+    public static final Serializer serializer = new Serializer();
+
+    public final ConstraintFunction function;
+    public final ColumnIdentifier columnName;
+
+    public final static class Raw
+    {
+        public final ConstraintFunction function;
+        public final ColumnIdentifier columnName;
+
+        public Raw(ColumnIdentifier functionName, ColumnIdentifier columnName)
+        {
+            this.columnName = columnName;
+            function = createConstraintFunction(functionName.toCQLString(), columnName);
+        }
+
+        public UnaryFunctionColumnConstraint prepare()
+        {
+            return new UnaryFunctionColumnConstraint(function, columnName);
+        }
+    }
+
+    private enum Functions
+    {
+        NOT_NULL(NotNullConstraint::new);
+
+        private final Function<ColumnIdentifier, ConstraintFunction> functionCreator;
+
+        Functions(Function<ColumnIdentifier, ConstraintFunction> functionCreator)
+        {
+            this.functionCreator = functionCreator;
+        }
+    }
+
+    private static ConstraintFunction createConstraintFunction(String functionName, ColumnIdentifier columnName)
+    {
+        try
+        {
+            return Functions.valueOf(LocalizeString.toUpperCaseLocalized(functionName)).functionCreator.apply(columnName);
+        }
+        catch (IllegalArgumentException ex)
+        {
+            throw new InvalidConstraintDefinitionException("Unrecognized constraint function: " + functionName);
+        }
+    }
+
+    private UnaryFunctionColumnConstraint(ConstraintFunction function, ColumnIdentifier columnName)
+    {
+        this.function = function;
+        this.columnName = columnName;
+    }
+
+    @Override
+    public MetadataSerializer<UnaryFunctionColumnConstraint> serializer()
+    {
+        return serializer;
+    }
+
+    @Override
+    public void appendCqlTo(CqlBuilder builder)
+    {
+        builder.append(toString());
+    }
+
+    @Override
+    public void evaluate(AbstractType<?> valueType, ByteBuffer columnValue) throws ConstraintViolationException
+    {
+        function.evaluate(valueType, columnValue);
+    }
+
+    @Override
+    public void validate(ColumnMetadata columnMetadata) throws InvalidConstraintDefinitionException
+    {
+        function.validate(columnMetadata);
+    }
+
+    @Override
+    public ConstraintType getConstraintType()
+    {
+        return UNARY_FUNCTION;
+    }
+
+    @Override
+    public String toString()
+    {
+        return function.getName() + '(' + columnName + ") ";
```

Comment:
Hah! Writing the tests I catched this bug. Look at the extra space on the closing parentheses. :-)

Don't worry about it. I'm fixing it on a PR to this PR.